ansible/roles/namespace-activator-scaled-objects/tasks/main.yml

```
@@ -0,0 +1,30 @@
+---
+
+  - name: 1pwd
+    kubernetes.core.k8s:
+      kubeconfig: ~/.kube/config
+      namespace: "{{ NAMESPACE }}"
+      template: keda-secret.yml.j2
+
+  - name: trigger auth
+    kubernetes.core.k8s:
+      kubeconfig: ~/.kube/config
+      namespace: "{{ NAMESPACE }}"
+      template: trigger-auth.yml.j2
+
+  - name: "generate scaled objects"
+    template:
+      src: scaled-object.yml.j2
+      dest: "{{ item }}-scaled-object.yml"
+    loop: "{{ deployments_for_scaled_objects }}"
+
+  - name: apply scaled objects
+    kubernetes.core.k8s:
+      kubeconfig: ~/.kube/config
+      namespace: "{{ NAMESPACE }}"
+      template: "{{ item }}-scaled-object.yml"
+      apply: true
+    loop: "{{ deployments_for_scaled_objects }}"
```

Why the split up and don't use the scaled-object.yml.j2 directly at the apply task?
The {{ item }} is also at the template who is use in apply scaled objects there.

```suggestion
  - name: apply scaled objects
    kubernetes.core.k8s:
      kubeconfig: ~/.kube/config
      namespace: "{{ NAMESPACE }}"
      template: scaled-object.yml.j2
      apply: true
    loop: "{{ deployments_for_scaled_objects }}"
```